homeassistant/components/unmanic/__init__.py

```diff
@@ -0,0 +1,115 @@
+"""The Unmanic integration."""
+import asyncio
+from datetime import timedelta
+import logging
+from typing import List
+
+from unmanic_api import Unmanic
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_VERIFY_SSL,
+    __version__,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, PLATFORMS
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup(hass: HomeAssistant, config: Config):
+    """Set up this integration using YAML is not supported."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up this integration using UI."""
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})
+
+    session = async_get_clientsession(hass)
+    client = Unmanic(
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        request_timeout=entry.data[CONF_TIMEOUT],
+        session=session,
+        tls=entry.data[CONF_SSL],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        user_agent=f"HomeAssistant/Unmanic/{__version__}",
+    )
+
+    coordinator = UnmanicUpdateCoordinator(hass, client=client)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    for platform in PLATFORMS:
+        if entry.options.get(platform, True):
```

Comment:
Why is this an option?

Reply:
Honestly, I'm not too sure why this was written as an option, but I used it from [integration_blueprint](https://github.com/custom-components/integration_blueprint/blob/8b3b9d380aac2b3473d5811b9450b9a412d5fe42/custom_components/integration_blueprint/__init__.py#L57-L62) and that's why it's there still. I tried changing this myself, but I couldn't get a working solution.

Reply:
What if you just remove it?

Reply:
If I remove it, all entities show as unavailable

Reply:
Wait, I forgot to import it and somehow didn't get an import error, this is all set